src/frequenz/sdk/timeseries/ev_charger_pool/_set_current_bounds.py

```diff
@@ -0,0 +1,120 @@
+
+import asyncio
+import logging
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Dict
+
+from frequenz.channels import Broadcast, Sender
+from frequenz.channels.util import Select, Timer
+
+from ..._internal.asyncio import cancel_and_await
+from ...microgrid import connection_manager
+from ...microgrid.component import ComponentCategory
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ComponentCurrentLimit:
+    """A current limit, to be sent to the ev charger."""
+
+    component_id: int
+    max_amps: float
+
+
+class BoundsSetter:
+    """A task for sending EV charger power bounds to the microgrid API."""
+
+    _NUM_PHASES = 3
+
+    def __init__(self, repeat_interval: timedelta) -> None:
+        """Create a `BoundsSetter` instance.
+
+        Args:
+            repeat_interval: Interval after which to repeat the last set bounds to the
+                microgrid API, if no new calls to `set_bounds` have been made.
+        """
+        self._repeat_interval = repeat_interval
+
+        self._task: asyncio.Task[None] = asyncio.create_task(self._run())
+        self._bounds_chan: Broadcast[ComponentCurrentLimit] = Broadcast("BoundsSetter")
+        self._bounds_rx = self._bounds_chan.new_receiver()
+        self._bounds_tx = self._bounds_chan.new_sender()
+
+    async def set(self, component_id: int, max_amps: float) -> None:
+        """Send the given current limit to the microgrid for the given component id.
+
+        Args:
+            component_id: ID of EV Charger to set the current bounds to.
+            max_amps: Current bound value to set for the EV Charger.
+        """
+        await self._bounds_tx.send(ComponentCurrentLimit(component_id, max_amps))
+
+    def new_bounds_sender(self) -> Sender[ComponentCurrentLimit]:
+        """Return a `Sender` for setting EV Charger current bounds with.
+
+        Returns:
+            A new `Sender`.
+        """
+        return self._bounds_chan.new_sender()
+
+    async def stop(self) -> None:
+        """Stop the BoundsSetter."""
+        await self._bounds_chan.close()
+        await cancel_and_await(self._task)
+
+    async def _run(self) -> None:
+        api_client = connection_manager.get().api_client
+        graph = connection_manager.get().component_graph
+        meters = graph.components(component_category={ComponentCategory.METER})
+        if not meters:
+            err = "No meters found in the component graph."
+            logger.error(err)
+            raise RuntimeError(err)
+
+        meter_data = (
+            await api_client.meter_data(next(iter(meters)).component_id)
+        ).into_peekable()
+        latest_bound: Dict[int, ComponentCurrentLimit] = {}
+
+        select = Select(
+            bound_chan=self._bounds_rx,
+            timer=Timer(self._repeat_interval.total_seconds()),
+        )
+        while await select.ready():
+            meter = meter_data.peek()
+
```

Comment:
you might want to check `if meter is None` here rather than withing each condition (if statement)